mode/php/php.js

```diff
@@ -111,6 +136,18 @@
           stream.eatWhile(/[\w\.]/);
           var delim = stream.current().slice(before[0].length + (quoted ? 2 : 1));
           if (quoted) stream.eat(quoted);
+
+          var mode;
+          if (!quoted && delim && delim === delim.toUpperCase() && (mode = getMode(delim))) {
+            state.delim = delim;
+            state.quoted = quoted;
+            state.curMode = mode;
```

Comment:
I think it'd be cleaner to put these three constant-over-the-whole-string fields in an object and store that in the state, so that there's less fields to reset later on (and less properties to copy when the state is copied)